app/mixins/event-wizard.js

```diff
@@ -210,6 +216,12 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
         this.sendAction('save');
       });
     },
+    savePublished() {
+      this.onValid(() => {
+        preSaveActions.call(this);
```

Comment:
Here savePublished does not set state to be published. Hence I understand, this is why is is savePublished